README.md

```diff
@@ -1,75 +1,96 @@
+
+Inbound secure WinRm network traffic (TCP port 5986) must be allowed from the GitHub Actions Runners virtual network so that remote sessions can be received.
+
+Prep the remote Windows server to accept WinRM management calls.  In general the Windows server needs to have a [WSMan] listener that looks for incoming [WinRM] calls. Firewall exceptions need to be added for the secure WinRM TCP ports, and non-secure firewall rules should be disabled. Here is an example script that would be run on the Windows server:
+
+  ```powershell
+  $Cert = New-SelfSignedCertificate -CertstoreLocation Cert:\LocalMachine\My -DnsName <<ip-address|fqdn-host-name>>
+
+  Export-Certificate -Cert $Cert -FilePath C:\temp\<<cert-name>>
+
+  Enable-PSRemoting -SkipNetworkProfileCheck -Force
+
+  # Check for HTTP listeners
+  dir wsman:\localhost\listener
+
+  # If HTTP Listeners exist, remove them
+  Get-ChildItem WSMan:\Localhost\listener | Where -Property Keys -eq "Transport=HTTP" | Remove-Item -Recurse
+
+  # If HTTPs Listeners don't exist, add one
+  New-Item -Path WSMan:\LocalHost\Listener -Transport HTTPS -Address * -CertificateThumbPrint $Cert.Thumbprint –Force
+
+  # This allows old WinRm hosts to use port 443
+  Set-Item WSMan:\localhost\Service\EnableCompatibilityHttpsListener -Value true
+
+  # Make sure an HTTPs inbound rule is allowed
+  New-NetFirewallRule -DisplayName "Windows Remote Management (HTTPS-In)" -Name "Windows Remote Management (HTTPS-In)" -Profile Any -LocalPort 5986 -Protocol TCP
+
+  # For security reasons, you might want to disable the firewall rule for HTTP that *Enable-PSRemoting* added:
+  Disable-NetFirewallRule -DisplayName "Windows Remote Management (HTTP-In)"
+  ```
+
+- `ip-address` or `fqdn-host-name` can be used for the `DnsName` property in the certificate creation. It should be the name that the actions runner will use to call to the Windows server.
+- `cert-name` can be any name.  This file will used to secure the traffic between the actions runner and the Windows server
 
 ## Example
 
 ```yml
-# TODO: Fill in the correct usage
+...
+env:
+  SERVICE_NAME: 'deploy-service'
+  SERVICE_PATH: 'c:\\services\\deploy'
+  WINDOWS_SERVER_SERVICE_USER: 'server_service_user'
+  WINDOWS_SERVER_SERVICE_PASSWORD: '${{ secrets.SERVER_SERVICE_SECRET }}'
+  WINDOWS_SERVER_CERT_PATH: './server-connection-cert.pfx'
+
+
 jobs:
-  job1:
-    runs-on: [self-hosted]
+  Deploy-Service:
+    runs-on: [windows-2019]
     steps:
-      - uses: actions/checkout@v2
+      ...
 
-      - name: Add the action here
-        uses: im-open/this-repo@v1.0.0
+      - name: Register Service
+        id: register
+        if: steps.deploy.outcome == 'success'
+        uses: im-open/register-windows-service@v1.0.0
         with:
-          input-1: 'abc'
-          input-2: '123
+          service-name: '${{ env.SERVICE_NAME }}'
+          deployment-path: '${{ env.SERVICE_PATH }}\\win-service.exe'
+          server: '${{ env.WINDOWS_SERVER }}'
```

Comment:
I did it this way, because this action will always be used in conjunction with other actions that need these pieces of information. I wanted to make the example relatively close to a real-world usage.

Reply:
In that case the `WINDOWS_SERVER` env doesn't exist in the list at the top.

Reply:
Sorry, the README didn't get added to the last push.  It's up to date now.